### DISCUSSION TITLE
Variables for commands

### DISCUSSION THREAD

**Discussion by ghost** [Zrips/CMI#3487] [Feature Requests]
**Description of feature/improvement request:**

I propose adding the @a, @e, @p, @r, @s variables to all CMI commands for player and entity.
Also if you can add the variables for them. Example: @a [distance = .. 5]; @e [type = minecraft: zombie]; @r [gamemode =! creative]; etc ...

---

**Comment by CloudeLecaw** [comment]
so where you suggest thus command where a Argument like this does not allready exists? Most commands have allready allplayer, playername, ir radius if needed. Most Cutsomalais and other things inside cmi suport the specials commands you find in wiki.and For command blocks these @ variables u wrote allready work if that is waht you want. So why this would be needed? 
Take a closer look here:
https://www.zrips.net/cmi/commands/

https://www.zrips.net/cmi/commands/specialized/

Also commands that need it like give or giveall Support metadatas (example: the Case you wanne spawn a Zombie with Special Feature, you wanne give an item with special stats lore name, enchantmenta) 

its never a good idea to have commandblock useage support in chatinput ommands.

**Comment by Zrips** [comment]
@Goddesu 
Those special variables require origin location, which in general can only be used from command block. But even then, command blocks already uses their own player detection by special variables you can add into it. CMI supports some what @p and @r variables when performing command from command block.
